docs/guides/implementation-guides/how-to-implement-dependency-injection.md

```diff
@@ -7,9 +7,120 @@ import MvvmArchitectureDiagram from '/img/gitbook-import/assets/image (3) (1) (2
 
 # How To Implement Dependency Injection
 
```

Comment:
Please add an introduction explaining what DI is and when you would use it. It would help greatly with people who have never heard of it before.

Reply:
I've copy/paste the introduction from the other PR, as it's seems ok.

Reply:
I would add an extra sentence somewhere, along the lines of "It accomplishes this by creating discrete services that are passed around/created as needed".

Reply:
Thanks, it's indeed a good improvement.